main.js

```diff
@@ -716,7 +727,110 @@ class Unifi extends utils.Adapter {
                 await this.applyJsonLogic(site, siteData, objects, this.statesFilter.dpi);
             }
         }
-    }    
+    }
+
+
+    /**
+     * Function to fetch daily gateway traffic
+     * @param {Object} sites 
+     */
+    async fetchGatewayTraffic(sites) {
+        return new Promise((resolve, reject) => {
+            this.controller.getDailyGatewayStats(sites, async (err, data) => {
+                if (err) {
+                    reject(new Error(err));
+                } else if (data === undefined || tools.isArray(data) === false || data[0] === undefined || tools.isArray(data[0]) === false) {
+                    reject(new Error('Returned data is not in valid format'));
+                } else {
+                    this.log.debug('fetchGatewayTraffic: ' + data[0].length);
+
+                    await this.processGatewayTraffic(sites, data);
+
+                    resolve(data);
+                }
+            }, undefined, undefined, ["lan-rx_bytes", "lan-tx_bytes"]);
+        });
+    }
+
+    /**
+     * Function that receives the daily gateway traffic as a JSON data array
+     * @param {Object} sites 
+     * @param {Object} data 
+     */
+    async processGatewayTraffic(sites, data) {
+        const objects = require('./admin/lib/objects_gatewayTraffic.json');
+
+        for (const site of sites) {
+            const x = sites.indexOf(site);
+
+            // Process objectsFilter
+            const siteData = data[x].filter((item) => {
+                // if (this.objectsFilter.dpi.includes(item.subsystem) !== true) {
+                //     return item;
+                // }
+                return item;
+            });
+
+            if (siteData.length > 0) {
+                await this.applyJsonLogic(site, siteData, objects, this.statesFilter.gatewayTraffic);
+            }
+        }
+    }
+
+    /**
+     * Function to fetch alarms
+     * @param {Object} sites 
+     */
+    async fetchAlarms(sites) {
+        return new Promise((resolve, reject) => {
+            this.controller.customApiRequest(sites, `/api/s/<SITE>/stat/alarm${this.update.alarmsNoArchived ? '?archived=false' : ''}`, async (err, data) => {
```

Comment:
Please add a note here that this call should be changed as soon as your PR in node-unifi (https://github.com/jens-maus/node-unifi/pull/27) will be integrated and you can use the standard API again.

Reply:
@jens-maus Until now I was not aware of custom API requests. Is there also the possibility to get not only online clients, but all clients incl. `last_seen_by_uap` and `last_seen_by_usw`? This would really simplifiy the calculation of `is_online` in ioBroker.